File: src/main/java/masonry/BasicBrick.java
package masonry;

import masonry.api.Brick;
import net.scriptgate.common.Point;

class BasicBrick implements Brick {
    private boolean isLayoutInstant = false;
    private Point size;
    private Point position = new Point(0, 0);

    public BasicBrick(int width, int height) {
        size = new Point(width, height);
    }

    @Override
    public void goTo(int x, int y) {
        this.position = new Point(x, y);
    }

    @Override
    public int getHeight() {
        return size.y;
    }

    @Override
    public int getWidth() {
        return size.x;
    }

    @Override
    public boolean isLayoutInstant() {
        return isLayoutInstant;
    }

    @Override
    public int getX() {
        return position.x;
    }

    @Override
    public int getY() {
        return position.y;
    }
}

File: src/main/kotlin/net/scriptgate/masonry/BasicBrick.kt
package net.scriptgate.masonry

import net.scriptgate.common.Point
import net.scriptgate.masonry.api.Brick

internal class BasicBrick(width: Int, height: Int) : Brick {
    override val isLayoutInstant = false
    private val size: Point
    private var position = Point(0, 0)

    override val height: Int
        get() = size.y

    override val width: Int
        get() = size.x

    override val x: Int
        get() = position.x

    override val y: Int
        get() = position.y

    init {
        size = Point(width, height)
    }

    override fun goTo(x: Int, y: Int) {
        this.position = Point(x, y)
    }
}
